src/us/kbase/narrativejobservice/sdkjobs/SDKLocalMethodRunner.java

```diff
@@ -461,6 +464,32 @@ public boolean isJobCanceled() {
                 log.logNextLine(resourceRequirements.toString(), false);
             }
 
+            //Get number of milliseconds to live for this token
+            //Set a timer before job is cancelled for having an expired token to the expiration time minus 10 minutes
+            //
+            final long msToLive = milliSecondsToLive(tokenStr, config);
+            Thread tokenExpirationHook = new Thread() {
+                @Override
+                public void run() {
+                    try {
+                        long tenMinutesBeforeExpiration = msToLive - 600000;
+                        if (tenMinutesBeforeExpiration > 0) {
```

Comment:
I would like this time to be a parameter that we can set.   So maybe "timeBeforeExpiration"

Reply:
done